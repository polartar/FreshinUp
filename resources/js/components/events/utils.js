import get from 'lodash/get'

const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}

export default getFileNameCopy
