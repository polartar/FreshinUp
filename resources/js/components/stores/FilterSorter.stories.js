import { storiesOf } from '@storybook/vue'
import { action } from '@storybook/addon-actions'
import MockAdapter from 'axios-mock-adapter'
import axios from 'axios/index'

// Components
import FilterSorter from './FilterSorter.vue'

let statuses = [
  { value: 1, text: 'Draft' },
  { value: 2, text: 'Pending' },
  { value: 3, text: 'Revision' },
  { value: 4, text: 'Rejected' },
  { value: 5, text: 'Approved' },
  { value: 6, text: 'On hold' }
]

let sortables = [
  { value: '-created_at', text: 'Newest' },
  { value: 'created_at', text: 'Oldest' },
  { value: 'title', text: 'Title (A - Z)' },
  { value: '-title', text: 'Title (Z - A)' }
]

const mock = new MockAdapter(axios)
mock.onGet('foodfleet/store-tags').reply(200, {
  data: [
    { uuid: 1, name: 'Store Tag 1' },
    { uuid: 2, name: 'Store Tag 2' },
    { uuid: 3, name: 'Store Tag 3' },
    { uuid: 4, name: 'Store Tag 4' }
  ]
})
mock.onGet('foodfleet/locations').reply(200, {
  data: [
    { uuid: 1, name: 'Location 1' },
    { uuid: 2, name: 'Location 2' },
    { uuid: 3, name: 'Location 3' },
    { uuid: 4, name: 'Location 4' }
  ]
})

storiesOf('FoodFleet|fleet-member/FilterSorter', module)
  .addParameters({
    backgrounds: [
      { name: 'default', value: '#f1f3f6', default: true }
    ]
  })
  .add('default', () => ({
    components: { FilterSorter },
    methods: {
      filterStores (params) {
        action('Run')(params)
      }
    },
    template: `
      <v-container>
        <filter-sorter
          @runFilter="filterStores"
        />
      </v-container>
    `
  }))
  .add('with statuses', () => ({
    components: { FilterSorter },
    data () {
      return {
        statuses: statuses
      }
    },
    methods: {
      filterStores (params) {
        action('Run')(params)
      }
    },
    template: `
      <v-container>
        <filter-sorter
          :statuses="statuses"
          @runFilter="filterStores"
        />
      </v-container>
    `
  }))
  .add('with sortables', () => ({
    components: { FilterSorter },
    data () {
      return {
        sortables: sortables
      }
    },
    methods: {
      filterStores (params) {
        action('Run')(params)
      }
    },
    template: `
      <v-container>
        <filter-sorter
          :sortables="sortables"
          @runFilter="filterStores"
        />
      </v-container>
    `
  }))
