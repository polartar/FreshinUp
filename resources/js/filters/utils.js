export function prefixStr (value, prefix = '@') {
  if (!value) return ''
  return value ? `${prefix} ${value}` : ''
}
