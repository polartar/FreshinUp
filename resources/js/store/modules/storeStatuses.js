import makeRestStore from 'fresh-bus/store/utils/makeRestStore'

export default ({ items, item }) => {
  return makeRestStore('foodfleet/store-statuses', { items, item })
}
