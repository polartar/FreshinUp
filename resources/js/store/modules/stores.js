import isObject from 'lodash/isObject'
import omitBy from 'lodash/omitBy'
import makeRestStore, {
  buildApi,
  makeModule
} from '@freshinup/core-ui/src/store/utils/makeRestStore'
import pick from 'lodash/pick'

export default ({ items, item }) => {
  const storesApi = buildApi('foodfleet/stores', { items, item })
  const store = makeModule(storesApi.getStore(), 'stores')

  const sortables = [
    { value: '-created_at', text: 'Newest' },
    { value: 'created_at', text: 'Oldest' },
    { value: 'name', text: 'Name (A - Z)' },
    { value: '-name', text: 'Name (Z - A)' }
  ]

  const FILLABLES = [
    'id',
    'uuid',
    'name',
    'square_id',
    'status_id',
    'type_id',
    'supplier_uuid',
    'address_uuid',
    'website',
    'contact_phone',
    'size',
    'image',
    'created_at',
    'updated_at',
    'deleted_at',
    'owner_uuid',
    'pos_system',
    'size_of_truck_trailer',
    'phone',
    'state_of_incorporation',
    'facebook',
    'twitter',
    'instagram',
    'staff_notes',
    'tags'
  ]

  // Initial State
  store.state = {
    ...store.state,
    sortables
  }

  const _createItem = store.actions.createItem
  const _updateItem = store.actions.updateItem
  const process = (payload) => {
    const data = pick(payload.data, FILLABLES)
    if (data.tags) {
      data.tags = data.tags.map(tag => {
        return isObject(tag) ? tag.uuid : tag
      })
    }
    return Object.assign({}, payload, { data })
  }
  store.actions = {
    ...store.actions,
    createItem (context, payload) {
      const processedPayload = process(payload)
      processedPayload.data = omitBy(processedPayload.data, (value) => {
        return String(value).length === 0
      })
      return _createItem(context, processedPayload)
    },
    updateItem (context, payload) {
      const processedPayload = process(payload)
      return _updateItem(context, processedPayload)
    }
  }

  return {
    namespaced: true,
    ...store,
    modules: {
      summary: makeRestStore(
        'summary',
        { item },
        {
          itemPath: ({ id }) => `/foodfleet/store-summary/${id}`
        }
      ),
      serviceSummary: makeRestStore(
        'serviceSummary',
        { item },
        {
          itemPath: ({ id }) => `/foodfleet/store-service-summary/${id}`
        }
      )
    }
  }
}
