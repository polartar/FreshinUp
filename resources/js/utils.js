import get from 'lodash/get'
/**
 *
 * @param options HTML attributes
 * @param [options.type] string 'link'|'script' or anything in fact
 * @param [options.container] HTMLElement
 * @returns {Promise<unknown>}
 */
export const lazyLoad = (options) => {
  return new Promise((resolve, reject) => {
    if (!options.type) {
      return reject(new Error('options.type is required'))
    }
    if (!options.container) {
      options.container = document.body
    }
    const { type, container, ...attributes } = options
    let element = document.createElement(type)
    Object.keys(attributes).forEach(key => element.setAttribute(key, attributes[key]))
    element.addEventListener('load', () => {
      resolve(element)
    })
    element.addEventListener('error', (e) => {
      reject(e)
    })
    container.append(element)
  })
}

export const getFileNameCopy = (name) => {
  const regex = /\s*\(([0-9]+)\)$/gm
  const matches = name.match(regex) || []
  const count = (
    parseInt(
      get(matches, '[0]', '')
        .replace('(', '')
        .replace(')', '')
    ) || 0
  ) + 1
  return `Copy of ${name.replace(get(matches, '[0]', ''), '')} (${count})`
}
