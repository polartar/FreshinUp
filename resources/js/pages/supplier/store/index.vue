
<script>
import stores from '~/pages/admin/fleet-members/index.vue'
import { mapGetters } from 'vuex'
const INCLUDE = [
  'tags',
  'addresses',
  'status',
  'owner',
  'type'
]

export default {
  extends: stores,
  computed: {
    ...mapGetters(['currentUser'])
  },
  beforeRouteEnterOrUpdate (vm, to, from, next) {
    vm.setPageLoading(true)
    Promise.all([
      vm.$store.dispatch('stores/setFilters', {
        ...vm.$route.query,
        ...this.lastFilterParams,
        owner_uuid: vm.currentUser.uuid
      }),
      vm.$store.dispatch('storeStatuses/getItems', {
        params: { include: INCLUDE }
      })
    ])
      .then(() => {
        if (next) next()
      })
      .catch((error) => console.error(error))
      .then(() => vm.setPageLoading(false))
  }
}
</script>
