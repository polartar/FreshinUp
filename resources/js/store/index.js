import { createStore } from 'fresh-bus/store'

import page from './modules/page'
import devices from './modules/devices'
import financialModifiers from './modules/financialModifiers'
import financialReports from './modules/financialReports'
import paymentTypes from './modules/paymentTypes'
import financialsummary from './modules/financialsummary'
import squares from './modules/squares'
import documents from './modules/documents'
import documentStatuses from './modules/documentStatuses'
import documentTypes from './modules/documentTypes'
import transactions from './modules/transactions'
import companyDetails from './modules/companyDetails'
import eventSummary from './modules/eventSummary'
import eventTypes from './modules/eventTypes'
import eventHistories from './modules/eventHistories'
import venues from './modules/venues'
import stores from './modules/stores'
import storeStatuses from './modules/storeStatuses'
import storeTypes from './modules/storeTypes'
import companies from './modules/companies'

export default (initialState = {}) => {
  return createStore(
    initialState,
    {
      modules: {
        page: page({}),
        devices: devices({}),
        financialModifiers: financialModifiers({}),
        financialReports: financialReports({}),
        paymentTypes: paymentTypes({}),
        financialsummary: financialsummary({}),
        squares: squares({}),
        documents: documents({}),
        documentStatuses: documentStatuses({}),
        documentTypes: documentTypes({}),
        transactions: transactions({}),
        companyDetails: companyDetails({}),
        eventSummary: eventSummary({}),
        eventTypes: eventTypes({}),
        eventHistories: eventHistories({}),
        venues: venues({}),
        stores: stores({}),
        storeStatuses: storeStatuses({}),
        storeTypes: storeTypes({}),
        companies: companies({})
      }
    }
  )
}
