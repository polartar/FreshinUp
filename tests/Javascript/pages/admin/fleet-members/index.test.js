import { shallowMount } from '@vue/test-utils'
import { createStore } from 'fresh-bus/store'
import { createLocalVue } from 'fresh-bus/tests/utils'
import { FIXTURE_STORES_RESPONSE, FIXTURE_STORES_SORTED_BY_FIRSTNAME } from 'tests/__data__/stores'
import Component from '~/pages/admin/fleet-members/index.vue'
import stores from '~/store/modules/stores'
import storeStatuses from '~/store/modules/storeStatuses'

describe('Admin Docs Page', () => {
  let localVue, mock, store, actions
  describe('Mount', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
    })

    afterEach(() => {
      mock.restore()
    })

    test('snapshot', async () => {
      const vue = createLocalVue({ validation: true })
      localVue = vue.localVue
      mock = vue.mock
        .onGet('api/foodfleet/stores', { params: { 'page[size]': 10, 'page[number]': 1, sort: 'first_name' } })
        .reply(200, { data: FIXTURE_STORES_SORTED_BY_FIRSTNAME })
        .onGet('api/foodfleet/stores').reply(200, FIXTURE_STORES_RESPONSE)
        .onAny()
        .reply(config => {
          console.warn('No mock match for ' + config.url, config)
          return [404, {}]
        })
      const store = createStore({}, {
        modules: {
          stores: stores({}),
          storeStatuses: storeStatuses({})
        }
      })
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })
      // Action: change State Machine's state
      await wrapper.vm.$store.dispatch('page/setLoading', false)
      await wrapper.vm.$nextTick()
      await wrapper.vm.$store.dispatch('stores/getItems')
      await wrapper.vm.$nextTick()
      expect(wrapper.element).toMatchSnapshot()
    })
  })

  describe('Methods', () => {
    beforeEach(() => {
      const vue = createLocalVue({ validation: true })
      const docModule = stores({})
      localVue = vue.localVue
      actions = {
        patchItem: jest.fn()
      }
      store = createStore({
        docs: {
          items: FIXTURE_STORES_RESPONSE
        }
      }, {
        modules: {
          stores: { ...docModule, actions: { ...docModule.actions, ...actions } }
        }
      })
    })

    afterEach(() => {
      mock.restore()
    })

    describe('storeViewOrEdit(store)', () => {
      const pushMock = jest.fn()
      const wrapper = shallowMount(Component, {
        localVue,
        store,
        mocks: {
          $router: {
            push: pushMock
          }
        }
      })
      wrapper.vm.storeViewOrEdit({ uuid: 'abc123' })
      expect(pushMock).toHaveBeenCalledWith({ path: `/admin/fleet-members/abc123/edit` })
    })

    test('changeStatus function change doc status', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.changeStatus(2, { uuid: 'mock uuid' })
      const data = { data: { status: 2 }, params: { id: 'mock uuid' } }
      expect(actions.patchItem).toHaveBeenCalled()
      expect(actions.patchItem.mock.calls).toHaveLength(1)
      expect(actions.patchItem.mock.calls[0][1]).toEqual(data)
    })

    test('changeStatusMultiple function change doc status for each', async () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.changeStatusMultiple(3, [{ uuid: 'mock uuid 1' }, { uuid: 'mock uuid 2' }])

      expect(actions.patchItem).toHaveBeenCalled()
      expect(actions.patchItem.mock.calls).toHaveLength(2)

      const firstData = { data: { status: 3 }, params: { id: 'mock uuid 1' } }
      const secondData = { data: { status: 3 }, params: { id: 'mock uuid 2' } }

      expect(actions.patchItem.mock.calls[0][1]).toEqual(firstData)
      expect(actions.patchItem.mock.calls[1][1]).toEqual(secondData)
    })

    test('onPaginate function change paginate', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.onPaginate({
        rowsPerPage: 2,
        totalItems: 5,
        page: 2
      })
      expect(wrapper.vm.pagination.rowsPerPage).toBe(2)
    })

    test('deleteStore function change deleteTemp', () => {
      const wrapper = shallowMount(Component, {
        localVue: localVue,
        store
      })

      wrapper.vm.deleteStore({ id: 1 })
      expect(wrapper.vm.deleteTemp[0].id).toBe(1)
      expect(wrapper.vm.deleteDialog).toBeTruthy()

      wrapper.vm.deleteStore([{ id: 1 }])
      expect(wrapper.vm.deleteTemp[0].id).toBe(1)
      expect(wrapper.vm.deleteDialog).toBeTruthy()
    })
  })
})
