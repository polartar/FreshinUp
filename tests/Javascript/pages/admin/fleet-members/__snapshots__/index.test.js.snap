// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pages/admin/fleet-members Mount snapshot 1`] = `
<div>
  <v-flex-stub
    align-center=""
    d-flex=""
    justify-space-between=""
    ma-2=""
    tag="div"
  >
    <h2
      class="white--text"
    >
      
      Fleet Member List
    
    </h2>
     
    <v-flex-stub
      tag="div"
      text-xs-right=""
    >
      <v-btn-stub
        activeclass="v-btn--active"
        color="primary"
        dark="true"
        href="/admin/fleet-members/new"
        slot="activator"
        tag="button"
        type="button"
      >
        
        Add New Fleet Member
      
      </v-btn-stub>
    </v-flex-stub>
  </v-flex-stub>
   
  <filter-sorter-stub
    filters="[object Object]"
    sortables="[object Object],[object Object],[object Object],[object Object]"
    statuses=""
  />
   
  <store-list-stub
    page="1"
    rowsperpage="10"
    rowsperpageitems="5,10,15,25,30,50"
    sortby=""
    statuses=""
    stores=""
    totalitems="0"
  />
   
  <v-dialog-stub
    contentclass=""
    maxwidth="500"
    origin="center center"
    transition="dialog-transition"
    width="auto"
  >
    <simple-confirm-stub
      cancellabel="No"
      class=""
      oklabel="Yes"
      title="Are you sure you want to delete this store?"
    >
      <div
        class="py-5 px-2"
      >
        <p
          class="subheading"
        >
          <span>
            Store
          </span>
          
            : 
          
        </p>
      </div>
    </simple-confirm-stub>
  </v-dialog-stub>
</div>
`;
